ProductWarehouse.Application/Contracts/IProductRepository.cs

```diff
@@ -0,0 +1,8 @@
+﻿using ProductWarehouse.Domain.Entities;
+
+namespace ProductWarehouse.Application.Contracts;
+
+public interface IProductRepository
+{
+    Task<IEnumerable<Product>> GetProductsAsync(decimal? minPrice, decimal? maxPrice, string? size);
```

Comment:
If you want this "decimal? minPrice, decimal? maxPrice, string? size" can be wrapped in new dto 
with the same name  ItemPriceDto or ItemsDto or something else. 

This is advice if you want it is not critical or bad. the decision is yours